### ISSUE TITLE
Flyout does not recieve DataContext correctly from attached element

### ISSUE THREAD

**Issue by takla21** [unoplatform/uno#10400]
### Current behavior

![ios](https://user-images.githubusercontent.com/17461593/201253433-3d74dfb6-7bdc-4e15-afbb-4b0ee709c43b.gif)

```cs
if (element is Button felement)
{
                FlyoutBase.SetAttachedFlyout(felement, Flyout);
                felement.Click -= Felement_Click;
                felement.Click += Felement_Click;
}

[..]

 private void Felement_Click(object sender, RoutedEventArgs e)
        {
            if (sender is FrameworkElement felement)
            {
                FlyoutBase.ShowAttachedFlyout(felement);
            }
        }
```

### Expected behavior

![uwp](https://user-images.githubusercontent.com/17461593/201253454-384bbd0b-173b-4b10-9952-ec2ecfb659d3.gif)

### How to reproduce it (as minimally and precisely as possible)

- Download this sample [FlyoutDataContext.zip](https://github.com/unoplatform/uno/files/9986176/FlyoutDataContext.zip)
- Deploy on any platform
- Click on any item

### Workaround

~~Manually setup the flyout DataContext~~
Workaround is not working actually

```
#if __ANDROID__ || __IOS__
Flyout.DataContext = item;
#endif
```

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

4.7.0-dev.146

### Affected platforms

Android, iOS

### IDE

Visual Studio 2022

### IDE version

17.3.5

**Comment by jeromelaban** [comment]
Thanks for the report!

**Comment by takla21** [comment]
We've found out that the workaround mention in the OP is actually not helping for this issue

**Comment by jeromelaban** [comment]
@takla21 this fix will be available in Uno 4.5.18, likely tomorrow.
